Add more realistic distribution of shear

trivialTZ opened the pull request:
add a Gaussian Mixture Model for gamma

trivialTZ committed:
Create gg_lensing_pop_calculation_tutorial.ipynb

trivialTZ committed:
Update gg_lensing_pop_calculation_tutorial.ipynb

add markdown cells to describe it.

trivialTZ committed:
Merge branch 'sibirrer:main' into main

trivialTZ committed:
Merge branch 'sibirrer:main' into main

trivialTZ committed:
Add more realistic distribution of shear

Using Gaussian Mixture Model (n=3) for gamma in gg_lens.py

trivialTZ committed:
Delete gg_lensing_pop_calculation_tutorial.ipynb

sibirrer commented on sim_pipeline/para_distribution.py:
```diff
@@ -0,0 +1,38 @@
+import numpy as np
+class GaussianMixtureModel:
```
can you make a test function of this class and their functions?
Also, perhaps make a separate folder within sim-pipeline called ParamDistributions and make a file called gaussian_mixture_model.py? (also add an __init__.py file to the folder). Then the code structure is better organized

sibirrer commented on sim_pipeline/gg_lens.py:
```diff
@@ -208,7 +208,12 @@ def los_linear_distortions(self):
         if not hasattr(self, '_gamma'):
-            gamma = np.random.normal(loc=0, scale=0.1)
+            mixture = GaussianMixtureModel(
+                means=[0.00330796, -0.07635054, 0.11829008],
```
there are many hard-coded values here. There is no information about from where these values are coming from. Perhaps put them in a configuration file (or class) with a description of what they are.
Also, the user (which might be you or someone else) might want to have different options. So you could define an input option (with a default setting).

sibirrer reviewed (changes requested):
See my comments. Excited to see how this is coming along!

trivialTZ committed:
Update related to gaussian mixture model

1. test function
2. ParamDistributions folder
3. add input option for gaussian mixture model

sibirrer reviewed (approved):
Thanks a lot!
For the future, you can also think of having a dedicated LineOfSight class that can organize all the options etc as things evolve

sibirrer committed:
Merge branch 'main' into main
